src/pages/index.tsx

```diff
@@ -1,10 +1,57 @@
 import Head from 'next/head';
 import { Inter } from '@next/font/google';
 import styles from '@/styles/Home.module.css';
+import { useEffect, useState } from 'react';
+import axios from 'axios';
 
 const inter = Inter({ subsets: ['latin'] });
 
+type GitHubRepository = {
+  description: string;
+  language: string;
+  name: string;
+  stars: number;
+  url: string;
+}
+
+type GitHubUserProfile = {
+  avatarUrl: string;
+  url: string;
+  bio: string;
+  name: string;
+}
+
+type GitHubProfile = {
+  languagesUsed: Map<string, number>;
+  profile?: GitHubUserProfile;
+  topFiveStarredRepos: GitHubRepository[];
+}
+
+type Profile = {
+  gitHub?: GitHubProfile;
+}
```

Comment:
We should probably move these `types` to a separate file (or files)?

Reply:
like a page just for types?